scene/gui/popup_menu.cpp

```diff
@@ -1457,6 +1457,7 @@ void PopupMenu::set_item_shortcut(int p_idx, const Ref<Shortcut> &p_shortcut, bo
 	items.write[p_idx].shortcut = p_shortcut;
 	items.write[p_idx].shortcut_is_global = p_global;
 	items.write[p_idx].dirty = true;
+	_shape_item(p_idx);
```

Comment:
Strange, this should not be necessary, setting `dirty` should be enough, since it's will reshape it on redraw.

Reply:
Oh, I see the issue was about menu position not drawing, so I guess a more universal solution would be adding shape to the size calculation, shortcut is likely not the only property that can cause size changes.

```diff
diff --git a/scene/gui/popup_menu.cpp b/scene/gui/popup_menu.cpp
index 10e13042a7..ca3bd8070f 100644
--- a/scene/gui/popup_menu.cpp
+++ b/scene/gui/popup_menu.cpp
@@ -60,6 +60,8 @@ Size2 PopupMenu::_get_contents_minimum_size() const {
 	for (int i = 0; i < items.size(); i++) {
 		Size2 size;
 
+		const_cast<PopupMenu *>(this)->_shape_item(i);
+
 		Size2 icon_size = items[i].get_icon_size();
 		size.height = _get_item_height(i);
 		icon_w = MAX(icon_size.width, icon_w);
```

Reply:
Yes I did originally do that as the fix, but the cost cast I wanted to avoid. If you think that is better I am happy to do that.